drv/cosmo-hf/src/main.rs

```diff
@@ -137,6 +132,35 @@ impl FlashDriver {
         let memory_type = bytes[1];
         let capacity = bytes[2];
 
+        // Make sure die 0 is selected with a dummy read, because the
+        // READ_UNIQUE_ID command is die-specific.
+        let mut buf = [0u8; 4];
+        self.flash_read(FlashAddr(0), &mut buf.as_mut_slice())
+            .unwrap_lite(); // infallible when given a slice
+        let die0_id = self.read_unique_id();
+
+        // Then read the top die's unique ID
+        self.flash_read(FlashAddr(0x04000000), &mut buf.as_mut_slice())
+            .unwrap_lite(); // infallible when given a slice
+        let die1_id = self.read_unique_id();
+
+        let mut unique_id = [0u8; 17];
```

Comment:
Why a 17 byte array when we're only storing 16 bytes?

Reply:
The `unique_id` array in `HfChipId` has to support multiple host flash types, so the _maximum_ unique ID length is 17 bytes (from the Gimlet's host flash).

Reply:
Gotcha. I guess the assumption that this and the packing format are clear to other readers? I don't think I would have made the connection to that size and this is where I would use some constant as a way to know that every callers has to change this and bump it rather than open-coding it everywhere. But maybe this is more straightforward due to a compile time error.

Reply:
I think if we add a _third_ flash type, then it would make sense to use enum variants, but right now, it's not too hard to make sure that the sides are in agreement about how we're using the byte array.